### ISSUE TITLE
Add a failure state for experiment install

### ISSUE THREAD

**Issue by johngruen** [mozilla/testpilot#660]

**Comment by johngruen** [comment]
![details cannot install 2x](https://cloud.githubusercontent.com/assets/3323249/17447567/1a92b9ca-5b1d-11e6-85a1-5de5e651d3b6.png)

**Comment by johngruen** [comment]
@wresuolc maybe we want to set a timeout limit here, like 10 seconds

**Comment by dannycoates** [comment]
There are two error states that we can easily add this message for, 'download-failed' and 'install-failed' but I don't think those alone will help all of the other issues linked to this one.

I was able to reproduce a hung download using the Network Link Conditioner from the [Hardware IO Tools for Xcode](https://developer.apple.com/download/more/) with the "Very Bad Network" setting.

The bad news is the hang didn't produce any errors in the 30 minutes I let it sit, until I finally turned off the link conditioner, refreshed the page, clicked enable again, and waited another 5 minutes. THEN I finally saw a "download-failed" message in the browser console -- followed by more "download-progress" and a successful install with no other action from me. 

So it doesn't seem like relying on an error message from the install listener is going to help much in the hung download case. A progress bar and maybe a cancel button would probably give people better feedback and control.

<img width="875" alt="screen shot 2016-08-08 at 3 24 56 pm" src="https://cloud.githubusercontent.com/assets/87619/17498614/4a175522-5d7d-11e6-9759-4d84554f837c.png">
